src/os/README.md

```diff
@@ -52,17 +52,19 @@ inspiration for several decisions in this library.
   add `OS\HErrnoException` if you want to report an error exposed via `h_errno`
 - add and use Hack classes (not type aliases) for 'handle'-like parameters and
   return values, e.g. `OS\open()` returns a `HH\Lib\FileDescriptor` instead of
-  an `int`; as well as aiding type safety, this prevents requests from interfering
-  with resources that belong to another request.
+  an `int`; as well as aiding type safety, this prevents requests from
+  interfering with resources that belong to another request.
 - Avoid `inout` parameters; return tuples instead. For example, prefer
   `function mkstemp(string $pattern): (FileDescriptor, string)` to
   `function mkstemp(inout string $in_pattern_out_path): FileDescriptor`
   - this can aid for common use by allowing string literals, rather than
     requiring otherwise-unused locals
-  - if the primary purpose of a function is to mutate its parameters, this
-    should be considered case-by-case; for example, if we were to add something
-    similar to `array_pop()`, it probably should take an `inout` container - but
-    probably shouldn't be in `OS\`
+  - if the primary purpose of a set of functions is to create, mutate and
+    destroy a C data structure, whose reference would not be held by C
+    libraries, the type, along with these utility functions, should be exposed
+    as a `vec` by default. See [Appendix: reference transparent C pointer
```

Comment:
Please either merge with #175, or re-add the deleted section and add 'list-like' or similar.

Reply:
> whose reference would not be held by C libraries

This is also *often* an implementation detail that is not even exposed to C users of the API.

Perhaps `Foo* foo_init()` just returns the new `Foo`, and trusts the app to call cleanup.

Or perhaps (even changing at runtime, e.g. depending on environment variables), it adds it to an internal list, so it can keep track of accounting metrics, or raise debug warnings if `_destroy()` is never called.

Overall I think this definition is extremely precise - but more precise than we can actually guarantee - and is overall a bad tradeoff vs readability

Reply:
Changed to:

> if the primary purpose of a set of functions is to create, mutate and
    destroy a C data structure that acts like a value instead of reference, the
    type, along with these utility functions, should be exposed as a `vec` or
    `shape`.